astra/src/test/java/com/slack/astra/chunk/IndexingChunkImplTest.java

```diff
@@ -646,6 +646,45 @@ public void testSnapshotToNonExistentS3BucketFails()
       assertThat(afterSearchNodes.get(0).url).contains(String.valueOf(TEST_PORT));
     }
 
+    @Test
+    public void testExcessFieldsAreDroppedButSpanIsIndexed() throws IOException {
+      int offset = 1;
+
+      // Create a span with too many fields (> 2500)
+      Trace.Span.Builder spanBuilder = SpanUtil.makeSpan(offset).toBuilder();
+      for (int i = 0; i < 3000; i++) {
+        spanBuilder.addTags(
+            Trace.KeyValue.newBuilder()
+                .setKey("custom.field." + i)
+                .setVStr("value" + i)
+                .setIndexSignal(Trace.IndexSignal.DYNAMIC_INDEX)
+                .build());
+      }
+      Trace.Span spanWithTooManyFields = spanBuilder.build();
+
+      // Add and commit the message
+      chunk.addMessage(spanWithTooManyFields, TEST_KAFKA_PARTITION_ID, offset);
+      chunk.commit();
+
+      // The message should be accepted, though some fields may be dropped
+      assertThat(getCount(MESSAGES_RECEIVED_COUNTER, registry))
+          .withFailMessage("Expected 1 message received")
+          .isEqualTo(1);
+      assertThat(getCount(MESSAGES_FAILED_COUNTER, registry))
+          .withFailMessage("Expected 0 messages failed")
+          .isEqualTo(0);
+      assertThat(getTimerCount(COMMITS_TIMER, registry))
+          .withFailMessage("Expected 1 commit recorded")
+          .isEqualTo(1);
+
+      long dynamicFieldsInSchema =
+          chunk.getSchema().keySet().stream().filter(key -> key.startsWith("custom.field")).count();
+
+      assertThat(dynamicFieldsInSchema)
+          .withFailMessage("Schema should not exceed 1500 fields but had %s", dynamicFieldsInSchema)
+          .isLessThanOrEqualTo(1500);
+    }
```

Comment:
Maybe add a test that builds a span with 3000 with dynamic fields and then 100 schema fields, and verify the schema fields are all present?